### PR TITLE
1. Added new new param:TunnelConnectionStatus in output object:VirtualNetworkGatewayConnection and 2. Added new optional input param:gatewayVip in ResetGateway API.

### PR DESCRIPTION
**Pull request by Nilambari**
This PR contains 2 changes :- 
- Added new param :- TunnelConnectionStatus in output Connection
object to show per tunnel connection health status. It is used in comment let:- Get-AzureRmVirtualNetworkGatewayConnection
- Added optional input param:- gatewayVip to pass gateway vip for ResetGateway API  [command let:- Reset-AzureRmVirtualNetworkGateway ]in case of Active
-Active feature enabled gateways.
- Updated helper file.

### CODE REVIEW

**Comment by Nilambari** [comment]
@markcowl Can you help accept this PR? Thanks a lot!

**Comment by Nilambari** [comment]
On demand:- http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1295/
Powershell Sign :- http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/678/

File: src/ResourceManager/Network/Commands.Network/Models/PSVirtualNetworkGatewayConnection.cs [hunk 1]
```diff
@@ -41,6 +42,8 @@ public class PSVirtualNetworkGatewayConnection : PSTopLevelResource
 
         public ulong IngressBytesTransferred { get; set; }
 
+        public List<PSTunnelConnectionHealth> TunnelConnectionStatus { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/Network/Commands.Network/Models/PSVirtualNetworkGatewayConnection.cs]: I don't see any test coverage for this functionality.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs [hunk 2]
```diff
@@ -32,6 +32,12 @@ public class ResetAzureVirtualNetworkGatewayCommand : VirtualNetworkGatewayBaseC
         [ValidateNotNull]
         public PSVirtualNetworkGateway VirtualNetworkGateway { get; set; }
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "The gateway vip in order to reset particular gateway instance (e.g. in case of Active-Active feature enabled gateways.) By default, gateway primary instance will be reset if no value is passed.")] 
+        public string GatewayVip { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs]: I don't see any test coverage for this.

File: src/ResourceManager/Network/Commands.Network/Models/PSVirtualNetworkGatewayConnection.cs (hunk 1 above)
Comment (Nilambari) [inline src/ResourceManager/Network/Commands.Network/Models/PSVirtualNetworkGatewayConnection.cs]: Updating test to cover this.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs (hunk 2 above)
Comment (Nilambari) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs]: Updating test to cover this.

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionSharedKeyCommand.cs [hunk 3]
```diff
@@ -44,7 +44,7 @@ public class ResetAzureVirtualNetworkGatewayConnectionSharedKeyCommand : Virtual
          ValueFromPipelineByPropertyName = true,
          HelpMessage = "The virtual network connection reset shared key length")]
         [ValidateNotNullOrEmpty]
-        public uint KeyLength { get; set; }
+        public int KeyLength { get; set; }
```
Comment (shahabhijeet) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/ResetAzureVirtualNetworkGatewayConnectionSharedKeyCommand.cs]: As we talked offline, please revert this change and update the PR

---
In reply to: [86872966](https://github.com/Azure/azure-powershell/pull/3170#discussion_r86872966) [](ancestors = 86872966)

File: src/ResourceManager/Network/ChangeLog.md [hunk 4]
```diff
@@ -19,10 +19,9 @@
 -->
 ## Current Release
 
-## Version 3.1.0
```
Comment (cormacpayne) [inline src/ResourceManager/Network/ChangeLog.md]: @Nilambari Hey Nilambari, please keep the information corresponding to release 3.1.0 in the change log so that customers looking at the change log can follow changes across multiple releases. Also, please move the information under "Version 3.2.0" to under "Current Release". At the end of the release, we run a script that will move the contents from under "Current Release" and put them under a newly created release section (in this case, 3.2.0).

File: src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs (hunk 2 above)
Comment (Nilambari) [inline src/ResourceManager/Network/Commands.Network/VirtualNetworkGateway/ResetAzureVirtualNetworkGatewayCommand.cs]: Done.

**Comment by Nilambari** [comment]
On demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/1296/

**Comment by Nilambari** [comment]
Can you Please help accept the PR? Thanks!
